src/lcos/barrier.cpp

```diff
@@ -59,6 +59,21 @@ namespace hpx { namespace lcos {
                 base_name, node_->get_unmanaged_id(), (*node_)->rank_).get();
     }
 
+    barrier::barrier(std::string const& base_name, std::vector<std::size_t> const& ranks, std::size_t rank)
+    {
+        auto rank_it = std::find(ranks.begin(), ranks.end(), hpx::get_locality_id());
```

Comment:
Shouldn't this use the `rank` parameter instead of `get_locality_id()`? Alternatively, could we get rid of the explicit `rank`?

Reply:
Good point, this should be `rank` indeed. I don't think we should get rid of the explicit `rank` parameter here.

Reply:
@sithhell Is there a use case where `rank != hpx::get_locality_id()`?

Reply:
Anything where you need a barrier on a topology that is not the same as the one of the localities.